src/devices/LedDisplay/Ht16k33.cs

```diff
@@ -0,0 +1,230 @@
+            /// </summary>
+            TurnOnOscillator = 0b0010_0001,
+            /// <summary>
+            /// HT16K33 set brightness command
+            /// </summary>
+            SetBrightness = 0b1110_0000,
+            /// <summary>
+            /// HT16K33 brightness bit mask
+            /// </summary>
+            BrightnessMask = 0b0000_1111
+        }
+        #endregion
+
+        /// <summary>
+        /// Display on/off state
+        /// </summary>
+        private bool _displayOn = true;
+
+        /// <summary>
+        /// Initial brightness level
+        /// </summary>
+        private byte _brightness;
+
+        /// <summary>
+        /// Initial blinking rate
+        /// </summary>
+        private BlinkRates _blinkRate;
+
+        /// <summary>
+        /// Send turn on oscillator command to device
+        /// </summary>
+        private void TurnOnOscillator() => I2cDevice.WriteByte((byte)Command.TurnOnOscillator);
+
+        /// <summary>
+        /// Send set brightness command to device
+        /// </summary>
+        private void SetBrightness() => I2cDevice.WriteByte((byte)((byte)Command.SetBrightness | (_brightness & (byte)Command.BrightnessMask)));
+
+        /// <summary>
+        /// Send Blink/DisplayOn command to device
+        /// </summary>
+        private void SetBlinkDisplayOn() => I2cDevice.WriteByte((byte)((byte)Command.Blink | (byte)Command.BlinkDisplayOn | ((byte)_blinkRate << 1)));
+        #endregion
+
+        #region Public members
+        #region Enums
+        /// <summary>
+        /// Blink rates
+        /// </summary>
+        /// <remarks>Specific to LED driver</remarks>
+        public enum BlinkRates : byte
+        {
+            /// <summary>
+            /// Turn off blinking
+            /// </summary>
+            Off = 0b00,
+            /// <summary>
+            /// Blink display at 2Hz (2x per second)
+            /// </summary>
+            Blink2Hz = 0b01,
+            /// <summary>
+            /// Blink display at 1Hz (1x per second)
+            /// </summary>
+            Blink1Hz = 0b10,
+            /// <summary>
+            /// Blink display at 0.5Hz (Once every 2 seconds)
+            /// </summary>
+            BlinkHalfHz = 0b11
+        }
+        #endregion
+
+        /// <summary>
+        /// Initialize HT16K33 device
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="System.Device.I2c.I2cDevice"/> to create with.</param>
+        public Ht16k33(I2cDevice i2cDevice)
+        {
+            // Initialize I2C device
+            I2cDevice = i2cDevice;
+
+            TurnOnOscillator();
+            SetBrightness();
+            SetBlinkDisplayOn();
+        }
+
+        /// <summary>
+        /// Gets or sets level of display brightness brightness 0..15
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Brightness must be between 0 and 15</exception>
+        public byte Brightness
```

Comment:
this abstraction feels weird because it requires for user to read the docs to use. I'd prefer if we exposed this as float/double from 0..1 which internally gets stored as byte (and reports back rounded value).

Presumably if users will allow to change this value frequently it will most likely be through UI with some kind of slider and I think float will make it easier to use.

Other alternative is to also add MaxBrightness const and make sure to use it in the sample (i.e. MaxBrightness / 2). That will be equally good.

Reply:
I like MaxBrightness better. I feel that using a float could result in rounding errors where you set a certain value and you get something different back if you read it and that might lead to some confusion.

Reply:
@gLes I don't mind either way. You pretty much always get different value back when you assign `0.1` to any variable